src/test/java/org/semux/consensus/SemuxSyncTest.java

```diff
@@ -206,4 +217,248 @@ public void testCheckpoints() {
         // tests
         assertFalse(sync.validateBlock(block, chain.getAccountState(), chain.getDelegateState()));
     }
+
+    @Test
+    public void testFastSync() throws Exception {
+        List<Key> keys = new ArrayList<>();
+        List<String> validators = new ArrayList<>();
+
+        for (int i = 0; i < 4; i++) {
+            keys.add(new Key());
+            validators.add(Hex.encode(keys.get(i).toAddress()));
+        }
+
+        BlockchainImpl chain = PowerMockito
+                .spy(new BlockchainImpl(kernelRule.getKernel().getConfig(), temporaryDBRule));
+        doReturn(validators).when(chain).getValidators();
+        PowerMockito.doNothing().when(chain, "updateValidators", anyLong());
+        kernelRule.getKernel().setBlockchain(chain);
+
+        SemuxSync sync = spy(new SemuxSync(kernelRule.getKernel()));
+        MessageQueue msgQueue = mock(MessageQueue.class);
+        Channel channel = mock(Channel.class);
+        when(channel.getRemoteAddress()).thenReturn(new InetSocketAddress("0.0.0.0", 5161));
+        when(channel.getMessageQueue()).thenReturn(msgQueue);
+
+        TreeSet<Pair<Block, Channel>> toProcess = new TreeSet<>(
+                Comparator.comparingLong(o -> o.getKey().getNumber()));
+
+        for (int i = 0; i < 200; i++) {
```

Comment:
This and the 199 below should probably at least be static constants (or config values later on).

'Magic numbers' are an anti-pattern.

There's nothing special about 200 afaik, just a convenient number for sync.

Reply:
Yes thank you. I was playing with the numbers and forgot to change it back to getValidatorUpdateInterval().